holoviews/core/dimension.py

```diff
@@ -151,10 +151,11 @@ def __init__(self, name, **params):
             all_params['name'] = name
         self.label = label
 
-        if not isinstance(params.get('values', None), basestring):
-            all_params['values'] = sorted(list(unique_array(params.get('values', []))))
-        elif params['values'] != 'initial':
-            raise Exception("Values argument can only be set with the string 'initial'.")
+        values = params.get('values', [])
+        if isinstance(values, basestring) and values == 'initial':
+            self.warning("The 'initial' string for dimension values is no longer supported.")
```

Comment:
Unless I'm confused set ``values`` to an empty list here, otherwise param will throw an exception after you've already warned.